ImageGoNord/GoNord.py

```diff
@@ -469,3 +473,232 @@ def save_image_to_file(self, image, path):
+        v_codec : str / optional
+            Video codec of the output
+
+        Returns
+        -------
+        None
+            Convert the numpy array to video and save to disk
+        """
+        
+        # If images is a list, convert to ndarray
+        if not isinstance(images, np.ndarray):
+            images = np.asarray(images)
+        height, width = images.shape[1:3]
+        process = (
+            ffmpeg
+                .input('pipe:', format='rawvideo', pix_fmt='rgb24', r=framerate, s='{}x{}'.format(width, height))
+                .output(fn, pix_fmt='yuv420p', vcodec=vcodec, loglevel='quiet')
+                .overwrite_output()
+                .run_async(pipe_stdin=True)
+        )
+        for idx, frame in enumerate(images):
+            self.clear_lines()
+            print(f'Frame: {start_frame + idx + 1}/{total_frames}')
+            process.stdin.write(
+                ConvertUtility.convert_palette(cube, frame)
+                    .astype(np.uint8)
+                    .tobytes()
+            )
+        process.stdin.close()
+        process.wait()
+
+    def concat_video(self, uid, out):
+        """
+        Concatenate two videos
+
+        Parameters
+        ----------
+        uid : str
+            Unique identifier for the session
+        out : str
+            Output video file path
+
+        Returns
+        -------
+        None
+            Concatenate two videos and save to disk
+        """
+        
+        main = ffmpeg.input(out)
+        temp = ffmpeg.input(f'temp_{uid}.mp4')
+        (
+            ffmpeg
+            .filter([main, temp],'concat')
+            .output(f'output_{uid}.mp4', pix_fmt='rgb24', loglevel='quiet')
+            .overwrite_output()
+            .run(capture_stdout=True)
+        )
+        os.remove(out)
+        os.remove(f'temp_{uid}.mp4')
+        os.rename(f'output_{uid}.mp4', out)
+        
+
+    def clear_lines(self, lines = 1):
+        """
+        Clear the last 'n' lines
+
+        Parameters
+        ----------
+        lines : int
+            Number of terminal lines to go up
+
+        Returns
+        -------
+        None
+            Clear lines in console
+        """
+
+        LINE_UP = '\033[1A'
+        LINE_CLEAR = '\x1b[2K'
+        for _ in range(lines):
+            print(LINE_UP, end=LINE_CLEAR)
+
+    def convert_video(self, _input, _output, _frames_per_batch = 200):
+        """
+        Concatenate two videos
+
+        Parameters
+        ----------
+        _input : str
+            Input video file path
+        _output : str
+            Output video file path
+
+        Returns
+        -------
+        None
+            Convert input video and save to disk
+        """
+        # Generate some random unique identifier that is generated for each session for the temporary files.
+        uid = uuid.uuid4()
+        palette = list(self.PALETTE_DATA.values())
+        #TODO: Find a fix for palette_name
```

Comment:
Is this still in todo?

Reply:
I'm thinking of adding the palette name as a parameter to the convert_video function itself. Will that be fine?  
The palette_name variable is actually required for name of the `.npz` file mentioned above.

Judging by the interaction with IGN website, it wouldn't be an issue for other palettes as with them it's not possible to deselect individual colors so we could add the palette folder name for them.